xdsl/dialects/printf.py

```diff
@@ -94,23 +94,31 @@ class PrintCharOp(IRDLOperation):
     """
     Print a single character
 
-    Equivalent to putchar in C.
+    Equivalent to putchar in C, but uses signless bytes as input (instead of ui32).
+    Unlike the C implementation, this op does not return anything.
     """
 
     name = "printf.print_char"
     char: Operand = operand_def(i8)
 
+    def __init__(self, char: SSAValue | Operation):
+        super().__init__(
+            operands=[char],
+        )
+
     @staticmethod
     def from_constant_char(char: str) -> PrintCharOp:
         """
         This constructor returns a PrintCharOp that prints the value supplied
         in "char" as a python char.
         """
-        assert len(char) == 1, "Only single characters are supported"
+        if len(char) != 1:
+            raise ValueError("Only single characters are supported")
```

Comment:
Characters is ambiguous, maybe something like this?
```suggestion
            raise ValueError(f'Unexpected char value "{char}", input must be a single ascii character')
```